js/runner.go

```diff
@@ -418,16 +418,23 @@ func (u *VU) RunOnce(ctx context.Context) error {
 	}
 
 	// Call the default function.
-	_, _, err := u.runFn(ctx, u.Runner.defaultGroup, u.Default, u.setupData)
+	_, isFullIteration, totalTime, err := u.runFn(ctx, u.Runner.defaultGroup, u.Default, u.setupData)
+
+	// If MinIterationDuration is specified and the iteration wasn't cancelled
+	// and was less than it, sleep for the remainder
+	if isFullIteration && u.Runner.Bundle.Options.MinIterationDuration.Valid {
+		time.Sleep(time.Duration(u.Runner.Bundle.Options.MinIterationDuration.Duration) - totalTime)
```

Comment:
Why did you remove the durationDiff variable and `if` check? With it, the code seems slightly more readable IMO, and it's very likely that the `time.Sleep()` call wouldn't be called in the majority of cases. While that's probably optimized to not create any timers actually do work for negative durations, it doesn't hurt us to have the check here. 

Edit: I saw the previous PR comment by @MStoykov (https://github.com/loadimpact/k6/pull/1175#pullrequestreview-294165478) and I respectfully disagree :smile: the 4 lines more clearly express what's happening and either have the same or slightly better performance.

Reply:
I'm not a fan of nested if, but agree that the original is much clearer for anyone, the later version requires knowledge of how `time.Sleep` works.